main.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/drone/drone-go/plugin/validator"
+
+	"github.com/kelseyhightower/envconfig"
+	log "github.com/sirupsen/logrus"
+)
+
+// plugin is fulfills the validator.Plugin interface
+type plugin struct{}
+
+// Validate approves builds from the same repository, and blocks builds from forks.
+func (p *plugin) Validate(ctx context.Context, req *validator.Request) error {
+	switch req.Build.Event {
+	case "push", "tag": // triggered by folks with write access to the repo, therefore trusted
+		log.Debugf("%s build ignored", req.Build.Event)
+		return nil
+	case "cron", "promote", "rollback", "custom": // triggered by folks with write access in drone
+		log.Debugf("%s build ignored", req.Build.Event)
+		return nil
+	case "pull_request": // may be triggered by folks without write access, needs approval
```

Comment:
If an external contributor pushes after submitting a pull request, is the event considered a push or a pull_request?

Reply:
It will show up as `pull_request`.  The example in the test file is a push:

https://github.com/gravitational/drone-fork-approval-extension/blob/c03dae11f8f11fad64bfd9241a42740c3b25b2a0/main_test.go#L21-L22

More info here:

https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#pull_request